### PR TITLE
Sprite fixes

### PR DESCRIPTION
**Pull request by BlueStaggo**
Moved player sprites up by one pixel and adjusted the HUD to be more accurate to the Genesis games.

image

### CODE REVIEW

**Comment by Techokami** [comment]
While I do appreciate the HUD fixes, are you *sure* about the player sprite adjustments? I could swear the player did, in fact, stand one pixel into the ground...

**Comment by DimensionWarped** [comment]
I feel like we're going to keep revisiting this subject over and over. Let's deal with this now...

Sonic 1 - in the ground
image

Sonic 1 - above the ground!
image

Sonic 1 tie breaker - above the ground
image

Sonic 2 - above the ground
image

Sonic 2 - above the ground
image

Sonic 3 - above the ground
image

Sonic 3 - above the ground
image

Above the ground wins.

File: Entities/MainObjects/Player.tscn [hunk 1]
```diff
@@ -72,7 +72,6 @@ resource_local_to_scene = true
 size = Vector2(18, 38)
 
 [sub_resource type="CanvasItemMaterial" id="2"]
-blend_mode = 5
```
Comment (DimensionWarped) [inline Entities/MainObjects/Player.tscn:75]: explain/revert

File: Entities/MainObjects/Player.tscn [hunk 2]
```diff
@@ -1294,13 +1293,14 @@ z_index = 6
 material = SubResource("2")
 sprite_frames = SubResource("57")
 animation = &"Insta"
-frame = 5
```
Comment (DimensionWarped) [inline Entities/MainObjects/Player.tscn:1297]: explain/revert

File: Entities/MainObjects/Player.tscn [hunk 3]
```diff
@@ -1341,11 +1341,11 @@ stream = ExtResource("58")
 bus = &"SFX"
 
 [node name="RingMagnet" type="Area2D" parent="."]
-visible = false
```
Comment (DimensionWarped) [inline Entities/MainObjects/Player.tscn:1344]: explain/revert

File: Entities/MainObjects/Player.tscn [hunk 4]
```diff
@@ -1341,11 +1341,11 @@ stream = ExtResource("58")
...
 collision_layer = 4096
 collision_mask = 0
 
 [node name="CollisionShape2D" type="CollisionShape2D" parent="RingMagnet"]
+visible = false
```
Comment (DimensionWarped) [inline Entities/MainObjects/Player.tscn:1348]: explain/revert

**Comment by DimensionWarped** [comment]
There are some changes that don't immediately appear relevant to the stated intention of the patch -- explain and/or revert them:

Acceptance Criteria
* Make sure the shift to the player's graphical mask doesn't look incorrect on any specific gimmick or terrain piece
* Make sure the change doesn't result in any mechanical issues
* Verify the UI looks right at all single minute digit times and double minute digit times.
* Verify that the UI looks right at single digit , two digit, and three digit ring counts.

File: Entities/MainObjects/Player.tscn (hunk 1 above)
Comment (BlueStaggo) [reply Entities/MainObjects/Player.tscn:75]: I can't access this value in the editor. Weird, but I'll revert this.

File: Entities/MainObjects/Player.tscn (hunk 2 above)
Comment (BlueStaggo) [reply Entities/MainObjects/Player.tscn:1297]: Just a random shield frame change. Reverted.

File: Entities/MainObjects/Player.tscn (hunk 3 above)
Comment (BlueStaggo) [reply Entities/MainObjects/Player.tscn:1344]: Ooh yup that shouldn't be visible. Reverted.

File: Entities/MainObjects/Player.tscn (hunk 4 above)
Comment (BlueStaggo) [reply Entities/MainObjects/Player.tscn:1348]: Opposite case with the Area2D. Reverted.

**Comment by BlueStaggo** [comment]
I will have to do a few changes because right now the upwards sprite shift is a little hacky (it uses transformations and I want to change it to use offsets, although this involves thoroughly looking through the animation player)

**Comment by BlueStaggo** [comment]
I have a feeling that my hacky solution of transforming the player sprite by -1 Y probably messes with things and needs checking up (I do not have the time and effort for this right now). Maybe it needs to be done using the offset instead?

**Comment by BlueStaggo** [comment]
I updated the hanging offsets but I think that may have messed up grabbing onto Tails, idk.

**Comment by DimensionWarped** [comment]
The Tails grabbing stuff is already so hacky that I'm not especially upset if it's a pixel off right now. It doesn't seem to have broken anything as far as I can tell testing at 60fps and 120fps, so I'm ok with this going in.

**Comment by DimensionWarped** [comment]
Went ahead and tested Amy and Knuckles on all gimmicks and they seem fine too.

**Comment by BlueStaggo** [comment]
Ugh, merge conflicts. I'm going to close this pull request but this should be easy (albeit a bit tedious) to recreate without the hacky method that I used.

**Pull request closed without merging**